src/components/rangeselector/draw.js

```diff
@@ -0,0 +1,273 @@
+
+        return (
+            axisLayout.range[0] === update[keys[0]] &&
+            axisLayout.range[1] === update[keys[1]]
+        );
+    }
+}
+
+function drawButtonRect(button, selectorLayout, d) {
+    var rect = button.selectAll('rect')
+        .data([0]);
+
+    rect.enter().append('rect')
+        .classed('selector-rect', true);
+
+    rect.attr('shape-rendering', 'crispEdges');
+
+    rect.attr({
+        'rx': constants.rx,
+        'ry': constants.ry
+    });
+
+    rect.call(Color.stroke, selectorLayout.bordercolor)
+        .call(Color.fill, getFillColor(selectorLayout, d))
+        .style('stroke-width', selectorLayout.borderwidth + 'px');
+}
+
+function getFillColor(selectorLayout, d) {
+    return (d.isActive || d.isHovered) ?
+        constants.activeColor :
+        selectorLayout.bgcolor;
+}
+
+function drawButtonText(button, selectorLayout, d) {
+    function textLayout(s) {
+        svgTextUtils.convertToTspans(s);
+
+        // TODO do we need anything else here?
+    }
+
+    var text = button.selectAll('text')
+        .data([0]);
+
+    text.enter().append('text')
+        .classed('selector-text', true)
+        .classed('user-select-none', true);
+
+    text.attr('text-anchor', 'middle');
+
+    text.call(Drawing.font, selectorLayout.font)
+        .text(getLabel(d))
+        .call(textLayout);
+}
+
+function getLabel(opts) {
+    if(opts.label) return opts.label;
+
+    if(opts.step === 'all') return 'all';
+
+    return opts.count + opts.step.charAt(0);
+}
+
+function reposition(gd, buttons, opts, axName) {
+    opts.width = 0;
+    opts.height = 0;
+
+    var borderWidth = opts.borderwidth;
+
+    buttons.each(function() {
+        var button = d3.select(this),
+            text = button.select('.selector-text'),
+            tspans = text.selectAll('tspan');
+
+        var tHeight = opts.font.size * 1.3,
+            tLines = tspans[0].length || 1,
+            hEff = Math.max(tHeight * tLines, 16) + 3;
+
+        opts.height = Math.max(opts.height, hEff);
+    });
+
+    buttons.each(function() {
+        var button = d3.select(this),
+            rect = button.select('.selector-rect'),
+            text = button.select('.selector-text'),
+            tspans = text.selectAll('tspan');
+
+        var tWidth = text.node() && Drawing.bBox(text.node()).width,
+            tHeight = opts.font.size * 1.3,
+            tLines = tspans[0].length || 1;
+
+        var wEff = Math.max(tWidth + 10, constants.minButtonWidth);
+
+        // TODO add MathJax support
+
+        // TODO add buttongap attribute
+
+        button.attr('transform', 'translate(' +
+            (borderWidth + opts.width) + ',' + borderWidth +
+        ')');
+
+        rect.attr({
+            x: 0,
+            y: 0,
+            width: wEff,
+            height: opts.height
+        });
+
+        var textAttrs = {
+            x: wEff / 2,
+            y: opts.height / 2 - ((tLines - 1) * tHeight / 2) + 3
```

Comment:
We should set `alignment-baseline="central"` so we can get more precise positioning here. 

[Docs](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/alignment-baseline) and [example](http://jsbin.com/caxivucamo/3/edit?html,js,output).